### PR TITLE
Add caching and additional discount information to Cart

### PR DESCRIPTION
**Pull request by glennjacobs**
Introduces caching of cart properties. This means we don't have to recalculate the cart totals when accessing via a different cart instance.

- [ ] Documentation updates
- [ ] Automated tests
- [ ] Changelog entries (core and hub)

### CODE REVIEW

File: packages/core/src/Models/Cart.php [hunk 1]
```diff
@@ -89,6 +144,18 @@ protected static function newFactory(): CartFactory
         'meta' => 'object',
     ];
 
+    /**
+     * The "booted" method of the model.
+     *
+     * @return void
+     */
+    protected static function booted()
```
Comment (alecritson) [inline packages/core/src/Models/Cart.php:152]: Can this not go on the trait? So you don't have to do it per model

```
public static function bootedCachesProperties();
```

Comment (glennjacobs) [reply packages/core/src/Models/Cart.php:152]: I think so. And it's also a reminder I've not even put it on the other models. 

I'll get that done and then start to make some tests for it.

**Comment by alecritson** [comment]
@glennjacobs Is this finished with? Be good to get it merged into `main` if ready.

**Comment by alecritson** [comment]
Closing as #676 has these changes in.

**Pull request closed without merging**